### ISSUE TITLE
Unable to build corso on Windows

### ISSUE THREAD

**Issue by meain** [alcionai/corso#2987]
In the [PR to add metrics](https://github.com/alcionai/corso/pull/2884), we seem to have introduced [`syscall.Kill`](https://github.com/alcionai/corso/blame/03edc603da3e6255405bebb4392b45d1018d28f0/src/internal/events/events.go#L238) which is not available on Windows. As of now, a temporary fix to avoid use of syscall on Windows is done via https://github.com/alcionai/corso/pull/2986 to unblock CI, but we have to properly fix this before next release.

An example failing run can be found [here](https://github.com/alcionai/corso/actions/runs/4560534347/jobs/8045919974#step:1:29).

**Comment by pandeyabs** [comment]
This is mitigated, although metrics on Windows doesn't work currently. I'll open a new issue for that.
